### PR TITLE
Adds a Guide on deploying Gradio apps with Docker

### PR DESCRIPTION
**Pull request by abidlabs**
When documenting the fix for https://github.com/gradio-app/gradio/issues/6920, I felt there was no adequate place to put this information in our existing docs/guides, so I added a Guide on deploying Gradio apps with Docker.

Closes: #6920 

fwiw this was requested quite a bit a long time ago (#378, #24) and I think would be good from a seo perspective

Also makes the exception that users observed in #6920 more informative

### CODE REVIEW

**Comment by abidlabs** [comment]
I'd also like to link to a section in the docs explaining all of the env variables that Gradio supports, but I'll do that once https://github.com/gradio-app/gradio/issues/6644 is in @aliabd

File: guides/09_other-tutorials/deploying-gradio-with-docker.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+iface = gr.Interface(fn=greet, inputs="text", outputs="text").launch()
+```
+
+This app creates a simple interface that greets the user by name.
+
+#### Step 2: Create a Dockerfile
+
+Next, we'll create a Dockerfile to specify how our app should be built and run in a Docker container. Create a file named `Dockerfile` in the same directory as your app with the following content:
+
+```dockerfile
```
Comment (aliabd) [inline guides/09_other-tutorials/deploying-gradio-with-docker.md:37]: Gonna add support for dockerfile syntax highlighting, right now this isn't rendering correctly

**Review by aliabd — APPROVED** [review]
LGTM @abidlabs. Made a small change to add dockerfile syntax highlighting on guides

**Comment by abidlabs** [comment]
Thanks for fixing it @aliabd!

**Pull request merged**